chore: deprecate axios (#1406)
nhost/nhost: .changeset/kind-cycles-explain.md

---
'@nhost/hasura-auth-js': minor
'@nhost/nhost-js': minor
---

Deprecate Axios

> Axios will be replaced by cross-fetch in the near future.

Reviewer: We are changing t
```suggestion
Axios will be replaced by cross-fetch in the near future. The signature of 2 methods is changing because we are making the API more flexible. 
```

> To prepare the transition, we recommend to adapt the following methods:

Reviewer: ```suggestion
To prepare for the transition, we are deprecating the old signature for the following methods:
```

- `nhost.functions.call()`
- `nhost.graphql.request()`

> Both methods are now marked as deprecated unless there is a specific `{ useAxios: false }` last parameter.

Reviewer: ```suggestion
Both methods now accept an optional `useAxios` parameter that can be used to opt-in (`{ useAxios: false }`) to the new method signature. By default, `useAxios` is set to `true` so you can update it on your own time.
```
